tests/pr_test.go

```diff
@@ -163,6 +163,57 @@ func TestRunUpgradeRoksPattern(t *testing.T) {
 	}
 }
 
+func TestValidateOverrideRoks(t *testing.T) {
+	t.Parallel()
+
+	options := setupOptionsRoksPattern(t, "slz-ovr-vld")
+	options.SkipTestTearDown = true
+	output, err := options.RunTestConsistency()
+
+	override_json := filepath.Join(roksPatternTerraformDir, "override.json")
+	config_json := options.LastTestTerraformOutputs["config"]
+	ignored_keys := []string{"resource_group"}
+
+	//Parse the json into a map
+	var override_json_map map[string]interface{}
+	err1 := json.Unmarshal([]byte(override_json), &override_json_map)
+	fmt.Println(err1)
+
+	config_json_byteArray, ok := config_json.([]byte)
+	if !ok {
+		fmt.Println("Error: Unable to convert config_json to []byte")
+		return
+	}
+
+	var config_map map[string]interface{}
+	err2 := json.Unmarshal([]byte(config_json_byteArray), &config_map)
+	fmt.Println(err2)
+
+	//Delete the ignored keys from the map
+	for _, key := range ignored_keys {
+		delete(override_json_map, key)
+		delete(config_map, key)
+	}
+
+	//Convert the maps back to json string
+	new_override_json_string_value, err3 := json.Marshal(override_json_map)
+	new_config_string_value, err4 := json.Marshal(config_map)
+
+	//Compare both the json strings
+	jsonComp, err3 := common.IsJsonEqual(string(new_override_json_string_value), string(new_config_string_value))
+
+	assert.Nil(t, err, "This should not have errored")
+	assert.NotNil(t, output, "Expected some output")
+	assert.NotNil(t, options.LastTestTerraformOutputs, "Expected some Terraform outputs")
+	assert.Nil(t, err1, "Error occured with the override json")
```

Comment:
You should probably handle assertions for errors immediately after they happen, usually when these errors occur you would want to mark the test as failed and then not continue with the test, since you will need the value that errored for the next step. See comments above on handling errors with "require" so that on critical errors the test will fail AND exit if required.

Reply:
Made the change